java/client/src/main/java/glide/api/commands/servermodules/FT.java

```diff
@@ -0,0 +1,164 @@
+package glide.api.commands.servermodules;
+
+import static glide.api.models.GlideString.gs;
+
+import glide.api.BaseClient;
+import glide.api.GlideClient;
+import glide.api.GlideClusterClient;
+import glide.api.models.ClusterValue;
+import glide.api.models.GlideString;
+import glide.api.models.commands.FT.FTCreateOptions;
+import glide.api.models.commands.FT.FTCreateOptions.FieldInfo;
+import java.util.Arrays;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import lombok.NonNull;
+
+/** Module for vector search commands. */
+public class FT {
+    /**
+     * Creates an index and initiates a backfill of that index.
+     *
+     * @param client The client to execute the command.
+     * @param indexName The index name.
+     * @param fields Fields to populate into the index.
+     * @return <code>OK</code>.
+     * @example
+     *     <pre>{@code
+     * // Create an index for vectors of size 2:
+     * FT.create(client, "my_idx1", new FieldInfo[] {
+     *     new FieldInfo("vec", VectorFieldFlat.builder(DistanceMetric.L2, 2).build())
+     * }).get();
+     *
+     * // Create a 6-dimensional JSON index using the HNSW algorithm:
+     * FT.create(client, "my_idx2",
+     *     new FieldInfo[] { new FieldInfo("$.vec", "VEC",
+     *         VectorFieldHnsw.builder(DistanceMetric.L2, 6).numberOfEdges(32).build())
+     * }).get();
+     * }</pre>
+     */
+    public static CompletableFuture<String> create(
+            @NonNull BaseClient client, @NonNull String indexName, @NonNull FieldInfo[] fields) {
+        // Node: bug in meme DB - command fails if cmd is too short even though all mandatory args are
+        // present
+        // TODO confirm is it fixed or not and update docs if needed
+        return create(client, indexName, fields, FTCreateOptions.builder().build());
+    }
+
+    /**
+     * Creates an index and initiates a backfill of that index.
+     *
+     * @param client The client to execute the command.
+     * @param indexName The index name.
+     * @param fields Fields to populate into the index.
+     * @param options Additional parameters for the command - see {@link FTCreateOptions}.
+     * @return <code>OK</code>.
+     * @example
+     *     <pre>{@code
+     * // Create a 6-dimensional JSON index using the HNSW algorithm:
+     * FT.create(client, "json_idx1",
+     *     new FieldInfo[] { new FieldInfo("$.vec", "VEC",
+     *         VectorFieldHnsw.builder(DistanceMetric.L2, 6).numberOfEdges(32).build())
+     *     },
+     *     FTCreateOptions.builder().indexType(JSON).prefixes(new String[] {"json:"}).build(),
+     * ).get();
+     * }</pre>
+     */
+    public static CompletableFuture<String> create(
+            @NonNull BaseClient client,
+            @NonNull String indexName,
+            @NonNull FieldInfo[] fields,
+            @NonNull FTCreateOptions options) {
+        return create(client, gs(indexName), fields, options);
```

Comment:
Will accepting String parameter (`indexName` here) but passing the GlideString version to customCommand() affect the output of the command anyhow?

Reply:
There is a binary version of custom command is used. I hope no.